frontend/lib/dev/example-static-page.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect } from "react";
+import { withRouter, RouteComponentProps } from "react-router-dom";
+import { getAppStaticContext } from "../app-static-context";
+
+type StaticPageProps = { children: JSX.Element };
+
+const StaticPage = withRouter(
+  (props: RouteComponentProps & StaticPageProps) => {
+    // If the user got here through a <Link> or other type of
+    // dynamic redirect, we want to trigger a page reload so
+    // that our static content is rendered server-side. However,
+    // we also want to ensure that if the user presses the
+    // back button on their browser, it triggers a hard redirect/reload
+    // instead of a popstate event (which won't do anything because
+    // we're a static page without any JS).  So we'll add a
+    // querystring to trigger this.
+    //
+    // Note that the following querystring should never actually
+    // be included in a <Link> on the site, or we'll break the user's
+    // browsing history.
+    useEffect(() => window.location.replace("?staticView"));
```

Comment:
Test this code.  Might have to use/modify the `hard-redirect.tsx` file for this.

Reply:
Ehh, the hard redirect is only really gonna be used during development anyways and it'd require refactoring `hard-redirect.tsx` so I'm not gonna deal with it.